packages/storybook/src/migrations/update-12-5-0/migrate-storybook-6-3.spec.ts

```diff
@@ -0,0 +1,203 @@
+            },
+          },
+          ['home-ui-angular']: {
+            projectType: 'library',
+            root: 'libs/home/ui-angular',
+            sourceRoot: 'libs/home/ui-angular/src',
+            targets: {
+              storybook: {
+                builder: '@nrwl/storybook:storybook',
+                options: {
+                  uiFramework: '@storybook/angular',
+                  port: 4400,
+                  config: {
+                    configFolder: 'libs/home/ui-angular/.storybook',
+                  },
+                },
+              },
+            },
+          },
+        },
+      });
+
+      const mainJSFile = `
+      module.exports = {
+        stories: [],
+        addons: ['@storybook/addon-knobs/register'],
+      };
+      `;
+
+      tree.write('.storybook/main.js', mainJSFile);
+      tree.write('libs/home/ui-angular/.storybook/main.js', mainJSFile);
+      tree.write('libs/home/ui-react/.storybook/main.js', mainJSFile);
+
+      writeJson(tree, 'package.json', {
+        devDependencies: {
+          '@storybook/addon-knobs': '6.3.0',
+        },
+      });
+    });
+
+    it('should update the main config files to fix the addon-knobs registration', async () => {
+      await updateStorybookv63(tree);
+
+      const globalMainJS = tree.read('.storybook/main.js', 'utf-8');
+      const angularMainJS = tree.read(
+        'libs/home/ui-angular/.storybook/main.js',
+        'utf-8'
+      );
+      const reactMainJS = tree.read(
+        'libs/home/ui-react/.storybook/main.js',
+        'utf-8'
+      );
+
+      const regex = /@storybook\/addon-knobs\/register/;
+      expect(regex.test(globalMainJS)).toBeFalsy();
+      expect(regex.test(angularMainJS)).toBeFalsy();
+      expect(regex.test(reactMainJS)).toBeFalsy();
+
+      const correctRegistrationRegex = /@storybook\/addon-knobs/;
+      expect(correctRegistrationRegex.test(globalMainJS)).toBeTruthy();
+      expect(correctRegistrationRegex.test(angularMainJS)).toBeTruthy();
+      expect(correctRegistrationRegex.test(reactMainJS)).toBeTruthy();
+    });
+  });
+
+  describe('package.json version upgrades', () => {
+    it('should update storybook packages', async () => {
+      writeJson(tree, 'package.json', {
+        devDependencies: {
+          '@storybook/angular': '^6.0.0',
+          '@storybook/react': '^6.0.0',
+          '@storybook/addon-knobs': '^6.0.0',
+          '@storybook/addon-controls': '^6.0.0',
+        },
+      });
+      await updateStorybookv63(tree);
+
+      expect(
+        readJson(tree, 'package.json').devDependencies['@storybook/angular']
+      ).toBe('~6.3.0');
+      expect(
+        readJson(tree, 'package.json').devDependencies['@storybook/react']
+      ).toBe('~6.3.0');
+      expect(
+        readJson(tree, 'package.json').devDependencies['@storybook/addon-knobs']
+      ).toBe('~6.3.0');
+      expect(
+        readJson(tree, 'package.json').devDependencies[
+          '@storybook/addon-controls'
```

Comment:
maybe we could change this to `@storybook/addon-essentials`, to work with the knobs-to-controls?